src/components/shared/ListView/ListView.tsx

```diff
@@ -0,0 +1,212 @@
+import {Box, ButtonGroup, Container, Grid, GridItem, Heading, IconButton, Text, VStack} from "@chakra-ui/react"
+import {useRouter} from "next/router"
+import {ListPage, ListPageWithFacets, Meta, MetaWithFacets, Product} from "ordercloud-javascript-sdk"
+import {ParsedUrlQuery} from "querystring"
+import {ReactElement, useCallback, useEffect, useMemo, useState} from "react"
+import {HiOutlineViewGrid, HiOutlineViewList} from "react-icons/hi"
+import {resourceLimits} from "worker_threads"
+
+interface ColumnDefinition<T> {
+  Header: string
+  accessor?: string
+  Cell?: ({row, value}: {row: {original: T}; value: any}) => ReactElement | string
+  sortable?: boolean
+}
+
+interface ListViewProps<T = {ID: string; Name: string}, F = any> {
+  service?: (...args) => Promise<T extends Product ? ListPageWithFacets<T, F> : ListPage<T>>
+  columns?: ColumnDefinition<T>
+  gridOptions?: {
+    templateColumns?: string
+    templateRows?: string
+    gap?: number
+  }
+  renderCard?: (
+    item: T,
+    index: number,
+    isSelected: boolean,
+    onSelectChange: (id: string, isSelected: boolean) => void
+  ) => ReactElement
+  paramMap?: {[key: string]: string}
+  queryMap?: {[key: string]: string}
+  children?: (props: ListViewChildrenProps) => React.ReactElement
+}
+
+export interface ListViewChildrenProps {
+  metaInformationDisplay: React.ReactElement
+  viewModeToggle: React.ReactElement
+  updateQuery: (queryKey: string) => (value: string | boolean | number) => void
+  routeParams: any
+  queryParams: any
+  children: any
+}
+
+const DEFAULT_CARD = (o: any, i: number) => (
+  <VStack
+    h="full"
+    justifyContent="space-between"
+    p={2}
+    backgroundColor="white"
+    border="1px solid"
+    borderColor="gray.200"
+    borderRadius="xl"
+    shadow="xl"
+  >
+    <Heading as="h3" fontSize="lg">
+      {o.Name.length > 39 ? o.Name.substring(0, 39) + "..." : o.Name}
+    </Heading>
+  </VStack>
+)
```

Comment:
What if instead of a default card, we just don't render the grid view if a card isn't passed in? I don't see a whole lot of value in providing a card that has just Name and would allow us to get rid of the `viewMode` prop

Reply:
I still think we need to allow both view modes, which will inherently require a `viewMode` prop to track what mode it's in. But I agree, we could say that the renderCard is optional and only return the toggleViewMode button when they provide a renderCard function.

Reply:
I'm going to keep the default render card, but it's specific to the DataGrid component. We can still make the ListView conditionally enable grid/table toggling but base it off of if gridOptions exist or not.